crates/aptos-inspection-service/src/server/memory_profiling.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright © Aptos Foundation
+
+use crate::server::utils::CONTENT_TYPE_TEXT;
+use hyper::{Body, StatusCode};
+use std::{path::Path, process::Command, thread, time::Duration};
+
+pub fn handle_memory_profiling_request() -> (StatusCode, Body, String) {
+    unsafe {
+        let mut prof_active: bool = true;
+
+        let result = jemalloc_sys::mallctl(
+            b"prof.active\0".as_ptr() as *const _,
+            std::ptr::null_mut(),
+            std::ptr::null_mut(),
+            &mut prof_active as *mut _ as *mut _,
+            std::mem::size_of::<bool>(),
+        );
+
+        println!("{}", result);
+        if result != 0 {
+            panic!("Failed to activate jemalloc profiling");
+        }
+
+        let handle = thread::spawn(move || {
+            // Sleep for 15 seconds
+            thread::sleep(Duration::from_secs(120));
+
+            // Disable the profiling
+            let mut prof_active: bool = false;
+            let result = jemalloc_sys::mallctl(
+                b"prof.active\0".as_ptr() as *const _,
+                std::ptr::null_mut(),
+                std::ptr::null_mut(),
+                &mut prof_active as *mut _ as *mut _,
+                std::mem::size_of::<bool>(),
+            );
+
+            println!("{}", result);
+            if result != 0 {
+                panic!("Failed to deactivate jemalloc profiling");
+            }
+        });
+
+        handle.join().unwrap();
+    }
+    let output = Command::new("python3")
+        .arg("/home/yunusozer/aptos-core/crates/aptos-inspection-service/src/server/memory_profile/jeprof.py")
```

Comment:
since `jeprof.py` is just two calls to subprocess, and this is one external sub command, may as well just do the two external subcommands here and delete `jeprof.py` ?

Reply:
Yes, for some reason it was erroring when I tried to do that so I just did it this way to temporarily unblock, but it's on my todo list. Thanks!